react_project/src/page/Market.jsx

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState } from 'react'
+import BestProduct from '../components/BestProduct'
+import AllProducts from '../components/AllProducts'
+import { getProducts } from '../data/api';
+
+
+export default function Market() {
+  const [order, setOrder] = useState('createdAt');
+  const [products, setProducts] = useState([]);
+  const sortedProducts = products.sort((a, b) => b[order] - a[order]);
+
+  const handleNewstClick = () => setOrder('createdAt');
+  const handleFavoriteClick = () => setOrder('favorite');
+  const handleLoad = async () => {
+  const { list } = await getProducts();
+    setProducts(list);
+  }
+
+  useEffect(() => {
+    handleLoad();
+  }, []);
+
+  return (
+    <>
+      <BestProduct products={sortedProducts} order='favorite' />
+      <AllProducts products={sortedProducts} onClick={[handleNewstClick, handleFavoriteClick]} />
```

Comment:
```js
export default function AllProducts({ products, handleNewstClick, handleFavoriteClick }) {
```
AllProducts가 위처럼 정의가 되어있어서 onClick이 아닌 아래처럼 구성이 되어야할 듯 합니다~
```suggestion
      <AllProducts products={sortedProducts} handleNewstClick={handleNewstClick} handleFavoriteClick ={handleFavoriteClick} />
```